app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt

```diff
@@ -162,4 +152,28 @@ class SectionHolder(
             start()
         }
     }
+
+    private fun showDescription() {
+        CustomDialog(
+            itemView.context,
+            viewModel.label(),
+            viewModel.description() ?: "",
+            itemView.context.getString(R.string.action_close),
+            null,
+            201,
+            null
+        ).show()
+    }
+
+    fun handleHeaderClick(x: Float) {
+        if (formBinding.descriptionIcon.visibility == View.VISIBLE) {
+            if (formBinding.descriptionIcon.x <= x && formBinding.descriptionIcon.x + formBinding.descriptionIcon.width >= x) {
+                showDescription()
+            } else {
+                onClick(itemView)
+            }
+        } else {
+            onClick(itemView)
+        }
```

Comment:
can we reduce it in a single If to avoid code nesting?

Reply:
done!